ingress/controllers/nginx/nginx/template.go

```diff
@@ -100,3 +107,68 @@ func toCamelCase(src string) string {
 	}
 	return string(bytes.Join(chunks, nil))
 }
+
+func buildLocation(input interface{}) string {
+	location, ok := input.(*Location)
+	if !ok {
+		return slash
+	}
+
+	path := location.Path
+	if len(location.Redirect.To) > 0 && location.Redirect.To != path {
+		return fmt.Sprintf("~* %s", path)
+	}
+
+	return path
+}
+
+func buildProxyPass(input interface{}) string {
```

Comment:
same as comment above, an example like: buildProxyPass produces the proxy pass string, if the ingress has redirects (specified through the kubernetes.io/... annotation), a base tag is added, eg: for an annotation like `kubernetes.io...` you will get: `<base href="http://">`...